### DISCUSSION TITLE
How to setup subdomain wildcard (e.g. *.subdomain.root.com)

### DISCUSSION THREAD

**Discussion by hnykda** [coollabsio/coolify#3453] [Q&A]
Hi. Coolify seems to work great with my A record being `*.domain.com` - adding `https://service.domain.com` to service domains works great.

I tried doing `*.sub.domain.com` (by adding CNAME to point to my coolify instance), but that doesn't seem to work - when I added assign `https://service.sub.domain.com`, I am getting the classic SSL error and I am not getting through. I checked the docs such as [Wildcard certificates](https://coolify.io/docs/knowledge-base/traefik/wildcard-certificates) but that seems to be something like if I want to dynamically generate those manually, which I don't - I want coolify's magic as with the simple case. I tried some approaches like setting the server's domain, and so on, but I apparently didn't find out the right switch. I otherwise have very-vanilla-mostly-default-settings, and stuff work.

Any hints or ideas? Thanks!

**Comment by hnykda** [comment]
Eh, I don't know what happened, but I just redeployed everything and it works now.

**Reply by sardonyx001** [reply]
Hi!
Can you please tell us more about how you set this up? 
I have a bunch of related apps and I want to deploy them all from the same docker compose file under the same subdomain: so they'll be like:
`docs.app.rootdomain.com`
`demo.app.rootdomain.com`
`blog.app.rootdomain.com`
etc..

do i just add a CNAME `*.app.rootdomain.com` for `rootdomain.com`?

**Reply by hnykda** [reply]
Yes, that should just work, I am using it. Doesn't it?

**Reply by sardonyx001** [reply]
I kept getting SSL errors while and then i read the cloudflare DNS [docs](https://developers.cloudflare.com/ssl/edge-certificates/additional-options/total-tls/error-messages/#:~:text=SSL/TLS%20certificate.-,By%20default%2C%20Cloudflare%20Universal%20SSL%20certificates%20only%20cover%20your%20apex%20domain,No,-To%20prevent%20insecure) and realized they dont automatically cover TLS for multilevel domains on the free plan :
<img width="520" alt="Screenshot 2024-10-17 at 20 39 10" src="https://github.com/user-attachments/assets/e7e401ac-2e85-4a96-81e6-6db415f8fcfa">

I will try again with caddy as a reverse proxy inside the containers since they offer on_demand certificates...

**Reply by hnykda** [reply]
I use just pure letsencrypt and a bare VM, so that might be the reason. That works just fine, tested on two different coolify instances and domains.

**Reply by sardonyx001** [reply]
Might be worth trying out letsencrypt.. Although caddy has it built it and (somehow) works with coolify although setting it up was a bit of a pain

**Comment by Rehaeel** [comment]
Hi guys.
As you said, I tried to do a follow-up, but nothing worked for me as you described. 

I have One NextJS app (`app.example.com`) on Coolify, and I installed Supabase for that instance (`db.app.example.com`).

1. The first flaw with Cloudflare with Coolify I see is that even if I add the wildcard domain `*.example.com` - every page auto-assigned in domain names - has no `www.*`. The Cloudflare recognizes it as a multi-level subdomain, as @Sardonyx001 wrote. My app is visited from multiple places where people still use www . (simple `www.app.example.com` doesn't work out from the box)
2. Adding the CNAME for my app, made no effect. Still having the mismatch error ([ERR_SSL_VERSION_OR_CIPHER_MISMATCH](https://developers.cloudflare.com/ssl/troubleshooting/version-cipher-mismatch/))

I don't understand what I should do to configure letsencrypt because the [docs](https://go-acme.github.io/lego/dns/cloudflare/) overwhelmed me. Can you point me to what exactly I should do to solve that multi-subdomain issue?
